apps/src/lib/node/ledger/mod.rs

```diff
@@ -1,425 +1,214 @@
-                    })?
-                }
-                AbciMsg::InitChain { reply, chain_id } => {
-                    self.init_chain(chain_id)?;
-                    reply.send(()).map_err(|e| {
-                        Error::AbciChannelSendError(format!("InitChain {}", e))
-                    })?
-                }
-                AbciMsg::MempoolValidate { reply, tx, r#type } => {
-                    let result = self
-                        .mempool_validate(&tx, r#type)
-                        .map_err(|e| format!("{}", e));
-                    reply.send(result).map_err(|e| {
-                        Error::AbciChannelSendError(format!(
-                            "MempoolValidate {}",
-                            e
-                        ))
-                    })?
-                }
-                AbciMsg::BeginBlock {
-                    reply,
-                    hash,
-                    height,
-                } => {
-                    self.begin_block(hash, height);
-                    reply.send(()).map_err(|e| {
-                        Error::AbciChannelSendError(format!("BeginBlock {}", e))
-                    })?
-                }
-                AbciMsg::ApplyTx { reply, tx } => {
-                    let (gas, result) = self.apply_tx(&tx);
-                    let result = result.map_err(|e| e.to_string());
-                    reply.send((gas, result)).map_err(|e| {
-                        Error::AbciChannelSendError(format!("ApplyTx {}", e))
-                    })?
-                }
-                AbciMsg::EndBlock { reply, height } => {
-                    self.end_block(height);
-                    reply.send(()).map_err(|e| {
-                        Error::AbciChannelSendError(format!("EndBlock {}", e))
-                    })?
-                }
-                AbciMsg::CommitBlock { reply } => {
-                    let result = self.commit();
-                    reply.send(result).map_err(|e| {
-                        Error::AbciChannelSendError(format!(
-                            "CommitBlock {}",
-                            e
-                        ))
-                    })?
-                }
-                AbciMsg::AbciQuery {
-                    reply,
-                    path,
-                    data,
-                    height: _,
-                    prove: _,
-                } => {
-                    if path == "dry_run_tx" {
-                        let result = self
-                            .dry_run_tx(&data)
-                            .map_err(|e| format!("{}", e));
-
-                        reply.send(result).map_err(|e| {
-                            Error::AbciChannelSendError(format!(
-                                "ApplyTx {}",
-                                e
-                            ))
-                        })?
+    fn call(&mut self, req: Request) -> Self::Future {
+        tracing::info!(?req);
```

Comment:
These are very verbose - for txs they print out the wasm code in bytes. Because info is on by default, we should only use it for things that are useful for users. You could maybe still print the request kind in debug

Reply:
:+1: